File: 2024_java_basics/12-while-loops-lab/tasks/07-moving.java
import java.util.Scanner;

public class Moving {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read the dimensions of the space
        int width = Integer.parseInt(scanner.nextLine());
        int length = Integer.parseInt(scanner.nextLine());
        int height = Integer.parseInt(scanner.nextLine());
        int homeSpace = width * length * height;

        String input = scanner.nextLine();

        while (!input.equals("Done")) {
            int box = Integer.parseInt(input);
            homeSpace -= box;

            if (homeSpace <= 0) {
                int diff = Math.abs(homeSpace);
                System.out.println("No more free space! You need " + diff + " Cubic meters more.");
                break;
            }

            input = scanner.nextLine();
        }

        if (homeSpace > 0) {
            System.out.println(homeSpace + " Cubic meters left.");
        }

        scanner.close();
    }
}

File: 2022_c#_basics/12-while-loop-exercise/07-moving.cs
using System;

namespace _07_Moving
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int w = int.Parse(Console.ReadLine());
            int l = int.Parse(Console.ReadLine());
            int h = int.Parse(Console.ReadLine());
            int homeSpace = h * l * w;
            string input = Console.ReadLine();

            while (input != "Done")
            {
                int box = int.Parse(input);
                homeSpace -= box;

                if (homeSpace <= 0)
                {
                    int dif = Math.Abs(homeSpace);
                    Console.WriteLine($"No more free space! You need {dif} Cubic meters more.");
                    break;
                }

                input = Console.ReadLine();
            }

            if (homeSpace > 0)
            {
                Console.WriteLine($"{homeSpace} Cubic meters left.");
            }
        }
    }
}
